core/resources/polygon.fs

```diff
@@ -2,11 +2,23 @@
 precision mediump float;
 #endif
 
-varying vec4 v_pos;
+uniform mat4 u_modelView;
+uniform mat4 u_modelViewProj;
+uniform float u_time;
+
+#pragma tangram: defines
+#pragma tangram: material
+#pragma tangram: _frag_lighting
+
 varying vec4 v_color;
+varying vec3 v_eyeToPoint;
 varying vec3 v_normal;
 varying vec2 v_texcoord;
 
 void main(void) {
-  	gl_FragColor = v_color;
+	vec4 color = v_color;
+
+	color *= calculateLighting(v_eyeToPoint,v_normal);
```

Comment:
Ideally the style shaders will be valid GL programs on their own, before any injection is performed. Without injecting the lighting code, compilation would fail on the line using the undefined `calculateLighting`.